src/filtering/predict.jl

```diff
@@ -77,7 +77,22 @@ function predict_cov!(
 
     _matmul!(view(R, 1:D, 1:D), Σ_curr.R, Ah')
     if !isone(diffusion)
-        _matmul!(view(R, D+1:2D, 1:D), Qh.R, sqrt.(diffusion))
+        apply_diffusion!(PSDMatrix(view(R, D+1:2D, 1:D)), Qh, diffusion)
+
+        # if diffusion isa Number
```

Comment:
When everything is tested and implemented to the degree that we want to in this PR, remove these.